### PR TITLE
Add build configuration for strictmode

### PR DESCRIPTION
**Pull request by aholstrup1**
#### Summary 
The PR sets up the needed scripts for running builds in StrictMode. We enable StrictMode by setting the StrictMode tag on a release branch. Once the tag is set we pick the latest available bcartifact of a given version as the baseline. 

#### Work Item(s) 
Fixes [AB#497503](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/497503)

### CODE REVIEW

File: build/scripts/PreCompileApp.ps1 [hunk 1]
```diff
@@ -55,6 +55,11 @@ if($appType -eq 'app')
                 }
             }
 
+            if($appBuildMode -eq 'Default' -and (Get-StrictModeBranchStatus)) {
+                # StrictMode tag is present on the branch so we build in StrictMode
+                $appBuildMode = "StrictMode"
```
Comment (mazhelez) [inline build/scripts/PreCompileApp.ps1]: Aren't both checks needed?

E.g. 24.1 is tested against latest released 24.1, as well as 24.0.
Strict mode shouldn't replace the _normal_ breaking changes checks.

Comment (aholstrup1) [reply build/scripts/PreCompileApp.ps1]: You're right 😄 AL-Go will need to support another build mode I think. I'll look into it

Comment (aholstrup1) [reply build/scripts/PreCompileApp.ps1]: Added this to AL-Go - Should be deployed to preview soon

File: build/scripts/GuardingV2ExtensionsHelper.psm1 [hunk 2]
```diff
@@ -43,6 +43,22 @@ function Enable-BreakingChangesCheck {
...
             $baselineVersion = $appJson.version # Use the version of the current app as the baseline version
         }
+        'Strict' {
+            Write-Host "Getting baseline for Strict Mode"
+            Import-Module -Name $PSScriptRoot\EnlistmentHelperFunctions.psm1
+
+            $majorMinor = Get-ConfigValue -Key "repoVersion" -ConfigType "AL-GO"
+            $strictModeVersion = ((Get-BCArtifactUrl -type Sandbox -country W1 -version $majorMinor -select Latest) -split "/")[-2]
+            if (-not $strictModeVersion) {
+                $strictModeVersion = ((Get-BCArtifactUrl -type Sandbox -country W1 -version $majorMinor -select Latest -storageAccount bcinsider -accept_insiderEula) -split "/")[-2]
```
Comment (mazhelez) [inline build/scripts/GuardingV2ExtensionsHelper.psm1]: Strict mode should only run on release branches, meaning that the baselines should also be released, so they should already be in _bcartifacts_ and not _bcinsider_.

File: build/scripts/GuardingV2ExtensionsHelper.psm1 [hunk 3]
```diff
@@ -205,4 +224,31 @@ function Update-AppSourceCopVersion
...
+function Get-StrictModeBranchStatus()
+{
+    Import-Module $PSScriptRoot\EnlistmentHelperFunctions.psm1
+    $StrictModeString = "/StrictMode"
+    $BranchName = Get-PullRequestTargetBranch
+    if (!$BranchName)
+    {
+        $BranchName = Get-CurrentBranch
+    }
+
```
Comment (mazhelez) [inline build/scripts/GuardingV2ExtensionsHelper.psm1:238]:
```suggestion
```

File: build/scripts/GuardingV2ExtensionsHelper.psm1 [hunk 4]
```diff
@@ -205,4 +224,31 @@ function Update-AppSourceCopVersion
     return $appSourceCopJsonPath
 }
 
+function Get-StrictModeBranchStatus()
```
Comment (mazhelez) [inline build/scripts/GuardingV2ExtensionsHelper.psm1]: I know where this comes for but consider giving it a better name that conveys "is strict mode enabled for branch".

**Review by mazhelez — APPROVED** [review]
Looks alright. Some code comments and doc could be useful.

**Review by mazhelez — DISMISSED** [review]

**Review by mazhelez — APPROVED** [review]

**Pull request merged**